lib/vsc/utils/script_tools.py

```diff
@@ -210,3 +226,178 @@ def critical_exception_handler(self, tp, value, traceback):
         message = "Script failure: %s - %s" % (tp, value)
         sys.exc_clear()
         self.critical(message)
+
+
+class CLI(object):
+    """
+    Base class to implement sync clients.
+    """
+    TIMESTAMP_MANDATORY = True
+
+    CLI_OPTIONS = {}
+    CACHE_DIR = "/var/cache"
+
+    def __init__(self, name=None, default_options=None):
+        """
+        Option
+            name (default: script name from commandline)
+            default_options: pass different set of default options
+                (only when creating a new parent class; for regular child classes, use CLI_OPTIONS)
+        """
+        if name is None:
+            name = _script_name(sys.argv[0])
+        self.name = name
+
+        self.fulloptions = self.make_options(defaults=default_options)
+        self.options = self.fulloptions.options
+
+        self.thresholds = None
+
+        self.start_timestamp = None
+        self.current_time = None
+
+    def make_options(self, defaults=None):
+        """
+        Take the default sync options, set the default timestamp file and merge
+        options from class constant OPTIONS
+
+        Return ExtendedSimpleOption instance
+        """
+        if defaults is None:
+            defaults = DEFAULT_CLI_OPTIONS
+        # use a copy
+        options = deepcopy(defaults)
+
+        # insert default timestamp value file based on name
+        if TIMESTAMP_FILE_OPTION in options:
+            tsopt = list(options[TIMESTAMP_FILE_OPTION])
+            tsopt[-1] = os.path.join(self.CACHE_DIR, "%s.timestamp" % self.name)
+            options[TIMESTAMP_FILE_OPTION] = tuple(tsopt)
+
+        options.update(self.CLI_OPTIONS)
+
+        if TIMESTAMP_FILE_OPTION not in options and self.TIMESTAMP_MANDATORY:
+            raise Exception("no mandatory %s option defined" % (TIMESTAMP_FILE_OPTION,))
+
+        return ExtendedSimpleOption(options)
+
+    def _update_kwargs_modified_since(self, kwargs):
+        """
+        Update dict kwargs in place with modified_since value based on start_timestamp and current_time
+        """
+        if 'modified_since' not in kwargs:
+            if self.start_timestamp is not None:
+                mtime = self.start_timestamp
+                if self.current_time is not None:
+                    mtime = '%s_%s' % (mtime, convert_to_unix_timestamp(self.current_time))
+                kwargs['modified_since'] = mtime
+
+    def warning(self, msg):
+        """
+        Convenience method that calls ExtendedSimpleOptions critical and exists with nagios warning exitcode
+        """
+        logging.warning(msg)
+        self.fulloptions.critical(msg)
+        sys.exit(NAGIOS_EXIT_WARNING[0])
```

Comment:
why not  use nagios exit with errocode  https://github.com/hpcugent/vsc-utils/blob/509c43c9a5e06d23fb00c83b8af24cd0377ea0f0/lib/vsc/utils/nagios.py#L137
then you can drop these functions?

or use a simplenagios object which has these functions?

Reply:
@stdweird ^ Any reason?